packages/core/src/vector/libsql/index.ts

```diff
@@ -278,6 +278,68 @@ export class LibSQLVector extends MastraVector {
     }
   }
 
+  /**
+   * Updates an index entry by its ID with the provided vector and/or metadata.
+   *
+   * @param indexName - The name of the index to update.
+   * @param id - The ID of the index entry to update.
+   * @param update - An object containing the vector and/or metadata to update.
+   * @param update.vector - An optional array of numbers representing the new vector.
+   * @param update.metadata - An optional record containing the new metadata.
+   * @returns A promise that resolves when the update is complete.
+   * @throws Will throw an error if no updates are provided or if the update operation fails.
+   */
+  async updateIndexById(
```

Comment:
These methods should exist in the base vector class and then we will need to add implementations for other vector stores. This can be done in another PR

Reply:
Did you mean this abstract class as the base vector class? I can add the method signatures there. 
https://github.com/mastra-ai/mastra/blob/aeb3bd4cc879a52d432ee9746f1c0acd1363d95d/packages/core/src/vector/index.ts#L17

Reply:
Yes

Reply:
Added the signatures to the abstract class.